### PR TITLE
DOCSP-41138: Distinct

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41138>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41138-distinct/read/distinct/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read/distinct.txt [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+   :values: reference
+ 
+.. meta::
+   :keywords: read, unique, code example
+
+Overview
+--------
+
+Within a collection, different documents might contain different values for a single field.
+For example, one ``restaurant`` document has a ``borough`` value of ``"Manhattan"``, and
```
Comment (jordan-smith721) [inline source/read/distinct.txt:24]: S:
```suggestion
For example, one document in the ``restaurant`` collection has a ``borough`` value of ``"Manhattan"``, and
```

File: source/read/distinct.txt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+The following {+language+} data class models the document in this collection:
```
Comment (jordan-smith721) [inline source/read/distinct.txt:37]:
```suggestion
The following {+language+} data class models the documents in this collection:
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM!

File: source/includes/read/distinct.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        .build()
+
+    val mongoClient = MongoClient.create(settings)
+    val database = mongoClient.getDatabase("sample_restaurants")
+    val collection = database.getCollection<Restaurant>("restaurants")
+
+    // start-distinct
+    val results = collection.distinct<String>(Restaurant::borough.name)
+
+    results.forEach { result ->
```
Comment (vbabanin) [inline source/includes/read/distinct.kt:30]: `forEach` appears to be deprecated according to the kotlin doc, which now recommends using `collect` instead. I'm unsure if the usage of `forEach` is specific to this PR or if we also need to review previously merged documentation for consistency. Could we verify if updates are needed elsewhere?

Comment (mcmorisi) [reply source/includes/read/distinct.kt:30]: Frankly, a lot of previously reviewed documentation in this repo has been using `forEach()`.

Comment (vbabanin) [reply source/includes/read/distinct.kt:30]: Thank you for clarifying, and my apologies for the confusion. You're correct; I mixed up the contexts between coroutines and synchronous implementations. `forEach` is indeed not deprecated for synchronous use in Kotlin, and collect is recommended specifically for asynchronous streams in coroutines.

**Review by vbabanin — APPROVED** [review]
LGTM!

**Pull request merged**